### DISCUSSION TITLE
Creating Global Strings

### DISCUSSION THREAD

**Discussion by Lonniebiz** [autokey/autokey#653] [Q&A]
I need more guidance than what's provided by the documentation, when it come to creating Global Strings:
https://askubuntu.com/questions/1387545/autokey-creating-a-global-variable

Where is the easiest place I can put a string, so that all my scripts can use that string?

**Comment by josephj11** [comment]
Welcome to AutoKey!

Start by taking a look [here](https://github.com/autokey/autokey/wiki/API-Examples#store). The rest of our wiki is also worth looking at. There are lots of examples and some explanations of the way things work. There's even an [example](https://github.com/autokey/autokey/wiki/Contributed-Scripts-2#toggle-a-looped-script) that uses global variables.

AutoKey offers saved variables and global saved variables. In both cases, it controls where they are stored and allows you to put, get, and delete their values using the provided API calls.

Since you want values to be available across different scripts, you need the global variables.

If you need anything more involved, you have the full power of Python available and can do things such as reading and writing external files which can contain arbitrarily complex data.

BTW, although posting questions like this here is fine, you will often get better results on our forums.

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.
